app/javascript/components/create-nuage-cloud-subnet-form.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import NuageCloudSubnetForm from './forms/nuage-cloud-subnet-form'
+import { createSubnet, fetchRouter } from '../utils/api.js'
+
+class CreateNuageCloudSubnetForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleFormStateUpdate = this.handleFormStateUpdate.bind(this);
+    this.state = {
+      emsId: false,
+      routerRef: false,
+      loading: true
+    }
+  }
+
+  componentDidMount() {
+    this.props.dispatch({
+      type: 'FormButtons.init',
+      payload: {
+        newRecord: true,
+        pristine: true,
+        addClicked: () => { createSubnet(this.state.values, this.state.emsId, this.state.routerRef); }
```

Comment:
You don't need the curly brackets around  `createSubnet(...)` function. You need them only for multi-line function body. Also you can drop the `;`